File: OOPS_Javascript/Abstraction.js
class Payment {
    processPayment() {
        throw new Error('Method not implemented.');
    }
}

class CreditCardPayment extends Payment {
    processPayment() {
        console.log('Processing credit card payment.');
    }
}

class PayPalPayment extends Payment {
    processPayment() {
        console.log('Processing PayPal payment.');
    }
}

const payment1 = new CreditCardPayment();
const payment2 = new PayPalPayment();

payment1.processPayment();
payment2.processPayment();

File: OOPS_Concepts/Python/Abstraction.py
# Abstraction in Python

from abc import ABC, abstractmethod

class Payment(ABC):
    @abstractmethod
    def process_payment(self):
        pass

class CreditCardPayment(Payment):
    def process_payment(self):
        print("Processing credit card payment.")

class PayPalPayment(Payment):
    def process_payment(self):
        print("Processing PayPal payment.")

payment1 = CreditCardPayment()
payment2 = PayPalPayment()

payment1.process_payment()
payment2.process_payment()
